### PR TITLE
Refactor to deal connection close

### PR DESCRIPTION
**Pull request by skrater**

### CODE REVIEW

File: amqp/connection.go [hunk 1]
```diff
@@ -8,75 +8,42 @@ import (
...
 // Close closes the AMQP connection.
 func (c *Connection) Close() {
-	c.channel.Close()
 	c.connection.Close()
 }
+
+func (c *Connection) WaitUntilConnectionClose() {
+	select {
+	case <-c.NotifyConnectionClose():
+	}
```
Comment (gfronza) [inline amqp/connection.go]: I believe we could take this `select` out and just go with:
```golang
<-c.NotifyConnectionClose()
```

File: README.md [hunk 2]
```diff
@@ -36,18 +36,22 @@ c.Subscribe("object.*", func(body []byte) bool {
     fmt.Println(body)
     return true
 })
+
+go c.Consume()
+
+conn.WaitUntilConnectionClose()
```
Comment (alexandrevicenzi) [inline README.md]: `WaitUntilConnectionCloses`

I think this is the correct english.

File: amqp/consumer.go [hunk 3]
```diff
@@ -19,19 +19,66 @@ type Consumer struct {
...
+	exchangeName string
+	queueName    string
 }
 
 // NewConsumer returns a new AMQP Consumer.
-func NewConsumer(c messaging.Connection, autoAck bool) (messaging.Consumer, error) {
+func NewConsumer(c messaging.Connection, autoAck bool, exchange, queue string) (messaging.Consumer, error) {
 	amqpConn := c.(*Connection)
 
+	ch, err := amqpConn.connection.Channel()
```
Comment (alexandrevicenzi) [inline amqp/consumer.go]: well, this is duplicated in producer class, not sure if there's a better solution.

Comment (skrater) [inline amqp/consumer.go]: i think, the best solution is receive the channel by parameter

@gfronza @alexandrevicenzi

Comment (alexandrevicenzi) [inline amqp/consumer.go]: maybe we could have a function called CreateChannel in the connection that do this and we pass the result to the constructor as a single parameter

Comment (alexandrevicenzi) [inline amqp/consumer.go]: or, channel could be a new class, like Producer and Consumer 😄

**Pull request merged**